src/Microsoft.Health.Api/Features/ByteCounter/ByteCountingStream.cs

```diff
@@ -0,0 +1,135 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Api.Features.ByteCounter
+{
+    /// <summary>
+    /// A stream to wrap an underlying stream and counts the number of bytes passed while operating with it.
+    /// </summary>
+    public class ByteCountingStream : Stream
```

Comment:
This probably makes sense in the "Core" project rather than API, right?

Reply:
I put it in API since it is meant to be used by dicom-server API project and added to the response body (see: https://github.com/microsoft/dicom-server/pull/1046)
But I guess it could be used by any kind of Stream so I can see why the "Core" project makes sense

Reply:
Moved stream to Core project. The HeaderUtility had to stay in Api project since it referenced AspNet, so I made it an extension instead